File: [프로그래머스] 연습문제_햄버거 만들기.py
def solution(ingredient):
    answer = 0
    s = []
    for i in ingredient:
        s.append(i)
        if s[-4:] == [1, 2, 3, 1]:  #인덱스 -4에서부터 끝까지
            answer += 1
            for _ in range(4):  #s 가장 뒤 원소 pop 4번 반복(1, 2, 3, 1 제거)
                s.pop()
    return answer

File: [프로그래머스] 연습문제_햄버거 만들기.js
function solution(ingredient) {
    var answer = 0;
    let arr = [];
    let str = ''
    for (let i of ingredient){
        arr.push(i)
        str = arr.slice(arr.length-4, arr.length).join('');
        if (arr.length >= 4 && str === '1231'){ //배열 그대로 비교할 수 없음(배열의 주소값을 비교함).따라서 문자열로 바꾸고 비교
            arr.splice(arr.length-4, 4); //.splice(-4)로 마지막 4개 삭제 가능
            answer++;
        }
    }
    return answer;
}